src/lib.rs

```diff
@@ -171,31 +200,42 @@ pub fn decode<T: ?Sized + AsRef<[u8]>>(input: &T) -> Result<Vec<u8>, DecodeError
 ///}
 ///```
 pub fn encode_config<T: ?Sized + AsRef<[u8]>>(input: &T, config: Config) -> String {
-    let mut buf = String::with_capacity(encoded_size(input.as_ref().len(), config));
+    let mut buf = String::with_capacity(encoded_size(input.as_ref().len(), &config));
 
     encode_config_buf(input, config, &mut buf);
 
     buf
 }
 
 /// calculate the base64 encoded string size, including padding
-fn encoded_size(bytes_len: usize, config: Config) -> usize {
+fn encoded_size(bytes_len: usize, config: &Config) -> usize {
+    let msg = "Encoded size exceeds usize";
     let rem = bytes_len % 3;
 
     let complete_input_chunks = bytes_len / 3;
-    let complete_output_chars = complete_input_chunks * 4;
-    let printing_output_chars = if rem == 0 {
-        complete_output_chars
+    let complete_chunk_output = complete_input_chunks.checked_mul(4).expect(msg);
```

Comment:
As Rust's support for 16 bit platforms broadens I'm trying to at least provide useful error messages rather than silently wrong results for tiny address space sizes.